### PR TITLE
all: adjust some build tags for wasi

### PR DESCRIPTION
**Pull request by bradfitz**
A start.

Updates #8320

### CODE REVIEW

File: cmd/derper/depaware.txt [hunk 1]
```diff
@@ -180,6 +180,7 @@ tailscale.com/cmd/derper dependencies: (generated by github.com/tailscale/depawa
         bytes                                                        from bufio+
         compress/flate                                               from compress/gzip+
         compress/gzip                                                from internal/profile+
+   L    compress/zlib                                                from debug/elf
```
Comment (bradfitz) [inline cmd/derper/depaware.txt:183]: These are due to a change in x/sys/unix (it depending on debug/elf now) and we need the newer x/sys/unix for wasip1 consts for josharian/native, etc.

Comment (bradfitz) [reply cmd/derper/depaware.txt:183]: I sent https://go-review.googlesource.com/c/sys/+/502356 to fix upstream.

File: types/logger/rusage_stub.go [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) Tailscale Inc & AUTHORS
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build windows || js
+//go:build windows || js || wasip1
```
Comment (DentonGentry) [inline types/logger/rusage_stub.go:4]: While harmless to leave here, the `js` build tag support in the filch files is being removed.

**Review by DentonGentry — APPROVED** [review]

File: types/logger/rusage_stub.go (hunk 2 above)
Comment (bradfitz) [reply types/logger/rusage_stub.go:4]: It's one of:

GOARCH=wasm GOOS=js
GOARCH=wasm GOOS=wasip1

So using build tag or filename with `wasm` effectively gets both, as those are the only two geese (GOOSes) with GOARCH=wasm.

**Pull request merged**